From f67044aebddb1555dc11d18786a71d78abe89be1 Mon Sep 17 00:00:00 2001
From: Jeff Morriss <jeff.morriss.ws@gmail.com>
Date: Tue, 4 Jun 2013 00:50:05 +0000
Subject: [PATCH 7/8] Fix the wiretap fuzz failure reported in
 https://bugs.wireshark.org/bugzilla/show_bug.cgi?id=8760
 :

Check that the record length we got out of the file is at least as big as
stats block trailer; if not, declare the file bad.

svn path=/trunk/; revision=49739
---
 wiretap/vwr.c |    8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/wiretap/vwr.c b/wiretap/vwr.c
index 8b7b8b9..a821652 100644
--- a/wiretap/vwr.c
+++ b/wiretap/vwr.c
@@ -713,8 +713,14 @@ static gboolean vwr_read(wtap *wth, int *err, gchar **err_info, gint64 *data_off
         return(FALSE);
     }
 
+    if (rec_size < (int)vwr->STATS_LEN) {
+        *err = file_error(wth->fh, err_info);
+        if (*err == 0)
+            *err_info = g_strdup_printf("vwr: Invalid record length %d (must be at least %u)", rec_size, vwr->STATS_LEN);
+            *err = WTAP_ERR_BAD_FILE;
+        return(FALSE);
+    }
     
-
     /* before writing anything out, make sure the buffer has enough space for everything */
     if ((vwr->FPGA_VERSION == vVW510021_W_FPGA) || (vwr->FPGA_VERSION == vVW510006_W_FPGA) )
     /* frames are always 802.11 with an extended radiotap header */
-- 
1.7.10.4

